sympy/printing/tests/test_maple.py

```diff
@@ -0,0 +1,413 @@
+from sympy.core import (S, pi, oo, symbols, Function, Rational, Integer,
+                        Tuple, Symbol, Eq, Ne, Le, Lt, Gt, Ge)
+from sympy.core import EulerGamma, GoldenRatio, Catalan, Lambda, Mul, Pow
+from sympy.functions import Piecewise, sqrt, ceiling, exp, sin, cos
+from sympy.utilities.pytest import raises
+from sympy.utilities.lambdify import implemented_function
+from sympy.matrices import (eye, Matrix, MatrixSymbol, Identity,
+                            HadamardProduct, SparseMatrix)
+from sympy.functions.special.bessel import (jn, yn, besselj, bessely, besseli,
+                                            besselk, hankel1, hankel2, airyai,
+                                            airybi, airyaiprime, airybiprime)
+from sympy.utilities.pytest import XFAIL
+
+from sympy import maple_code
+
+x, y, z = symbols('x,y,z')
+
+
+def test_Integer():
+    assert maple_code(Integer(67)) == "67"
+    assert maple_code(Integer(-1)) == "-1"
+
+
+def test_Rational():
+    assert maple_code(Rational(3, 7)) == "3/7"
+    assert maple_code(Rational(18, 9)) == "2"
+    assert maple_code(Rational(3, -7)) == "-3/7"
+    assert maple_code(Rational(-3, -7)) == "3/7"
+    assert maple_code(x + Rational(3, 7)) == "x + 3/7"
+    assert maple_code(Rational(3, 7) * x) == '(3/7)*x'
+
+
+def test_Relational():
+    assert maple_code(Eq(x, y)) == "x = y"
+    assert maple_code(Ne(x, y)) == "x <> y"
+    assert maple_code(Le(x, y)) == "x <= y"
+    assert maple_code(Lt(x, y)) == "x < y"
+    assert maple_code(Gt(x, y)) == "x > y"
+    assert maple_code(Ge(x, y)) == "x >= y"
+
+
+def test_Function():
+    assert maple_code(sin(x) ** cos(x)) == "sin(x)^cos(x)"
+    assert maple_code(abs(x)) == "abs(x)"
+    assert maple_code(ceiling(x)) == "ceil(x)"
+
+
+def test_Pow():
+    assert maple_code(x ** 3) == "x^3"
+    assert maple_code(x ** (y ** 3)) == "x^y^3"
+
+    print(maple_code((x ** 3) ** y))
+    assert maple_code((x ** 3) ** y) == "(x^3)^y"  # PROBLEM
```

Comment:
You can use `srepr` to see the resulting representation:
```
In [6]: srepr((x**3)**y)
Out[6]: "Pow(Pow(Symbol('x', integer=True), Integer(3)), Symbol('y'))"

In [7]: srepr(x**(3**y))
Out[7]: "Pow(Symbol('x', integer=True), Pow(Integer(3), Symbol('y')))"
```

Hence, `_print_Pow` could use a parenthesize function on both arguments. I would look at e.g. the LaTeX printer:

```
    def parenthesize(self, item, level, strict=False):
        prec_val = precedence_traditional(item)
        if (prec_val < level) or ((not strict) and prec_val <= level):
            return r"\left({}\right)".format(self._print(item))
        else:
            return self._print(item)
```
and replace the line `return r"\left({}\right)".format(self._print(item))` with `return "({})".format(self._print(item))`.

Then, in `_print_Pow`
```
 base = self.parenthesize(expr.base, PRECEDENCE['Pow'])
```
where you can import `PRECEDENCE` by
```
from sympy.printing.precedence import PRECEDENCE
```